### PR TITLE
Add Functions for Cardano Utilities APIs with Unit and Integration Tests

### PR DESCRIPTION
**Pull request by hieu-dd**
# Pull Request (PR) Summary

- This PR introduces new functions for accessing Cardano Utilities APIs in the SDK, along with comprehensive unit and integration tests.

## Changes Description

In this PR, I have added a series of new functions to the SDK that provide access to various Cardano Utilities APIs. These APIs include tools and services that are essential for developers working with the Cardano blockchain, such as address validation, transaction estimation, and other utility functions.

To ensure the reliability and functionality of these new functions, I have implemented both unit tests and integration tests. These tests are designed to verify that the functions perform correctly under various scenarios and handle both typical and edge cases effectively.

## Related Issue(s)

- Fixes # (issue number)  // Please replace with actual issue number if applicable

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Other (please describe)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] My PR is targeting the `main` branch
- [x] I have added necessary documentation (if appropriate)
- [x] I have added tests that prove my fix is effective or my feature works
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/model/TransactionPayload.kt [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+/**
+ * TxOutValue
+ *
+ * @property coins Coins
+ * @property assets Assets
+ */
+@Serializable
+data class TxOutValue(
+    val coins: Int,
+    val assets: Map<String, Int>,
```
Comment (iTanChi) [inline core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/model/TransactionPayload.kt]: Mấy cái này nó là số big đó, coin và asset value

Comment (iTanChi) [reply core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/model/TransactionPayload.kt]: zcheck lại các api khác nữa nhé

File: core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/model/TransactionPayload.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+)
+
+@Serializable
+data class TransactionPayload(
+    val cbor: String,
+    @Serializable(with = UtxoTupleListSerializer::class)
+    val additionalUtxoSet: List<List<UtxoTuple>>,
```
Comment (iTanChi) [inline core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/model/TransactionPayload.kt]: này sao nó là list của list nhỉ

Comment (hieu-dd) [reply core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/model/TransactionPayload.kt]: data nó thế
```
{
  "cbor": "string",
  "additionalUtxoSet": [
    [
      {
        "txId": "string",
        "index": 0
      },
      {
        "address": "string",
        "value": {
          "coins": 0,
          "assets": {}
        },
        "datum_hash": "string",
        "datum": {},
        "script": {}
      }
    ]
  ]
}
```

Comment (iTanChi) [reply core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/model/TransactionPayload.kt]: Đổi về object đi, đặt tên là UTxO có input là phần tử đầu tiên và output là phần tử thứ 2:
``` 
data class UTxO {
    val input: TransactionInput,
    val output: TransactionOutput
}
```

Comment (hieu-dd) [reply core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/model/TransactionPayload.kt]: đổi thằng nào nhỉ?
```
 {
        "txId": "string",
        "index": 0
      },
      {
        "address": "string",
        "value": {
          "coins": 0,
          "assets": {}
        },
        "datum_hash": "string",
        "datum": {},
        "script": {}
      }
```
này à. Nma như thế lsao nó parse được nhỉ? hay cũng phải viết custom seriable

File: core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/model/TransactionPayload.kt [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+/**
+ * Transaction Payload
+ *
+ * @property cbor CBOR
+ * @property uTxO UTxO
+ */
+data class TransactionPayload(
+    val cbor: String,
+    val uTxO: UTxO? = null,
```
Comment (iTanChi) [inline core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/model/TransactionPayload.kt]: cái này phải là list chứ

File: core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/CardanoUtilitiesApi.kt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        xpub: String,
+        role: Int,
+        index: Int,
+    ) = client.fetchResource<DerivedAddress>(
+        PATH_DERIVE_ADDRESS.replace(":xpub", xpub)
+            .replace(":role", role.toString())
+            .replace(":index", index.toString()),
+    )
+
+    suspend fun submitTransactionForExecutionUnitsEvaluation(transaction: String) =
```
Comment (iTanChi) [inline core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/CardanoUtilitiesApi.kt:26]: tên là transactionCbor nha

File: core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/model/TransactionPayload.kt [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+)
+
+/**
+ * Transaction Payload
+ *
+ * @property cbor CBOR
+ * @property uTxO UTxO
+ */
+data class TransactionPayload(
+    val cbor: String,
```
Comment (iTanChi) [inline core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/model/TransactionPayload.kt]: tên là transactionCbor nha, bên dưới là additionalUtxoSet, những chỗ khác có tên tương tự thì rename luôn nhé

File: core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/model/TransactionPayload.kt [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+ * @property value Value
+ * @property datumHash Datum hash
+ * @property datum Datum
+ * @property script Script
+ */
+data class TransactionOutput(
+    val address: String,
+    val value: TransactionOutputValue,
+    val datumHash: String,
+    val datum: Map<String, JsonElement>,
```
Comment (iTanChi) [inline core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/model/TransactionPayload.kt]: Chỗ này chắc là phải check lại là dùng một module Cardano Model để dùng lại thôi

File: core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/model/TransactionPayload.kt [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+ * @property cbor CBOR
+ * @property uTxO UTxO
+ */
+data class TransactionPayload(
+    val cbor: String,
+    val uTxO: UTxO? = null,
+)
+
+private fun TransactionPayload.toPayload() =
+    JsonObject(
```
Comment (iTanChi) [inline core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/utilities/model/TransactionPayload.kt]: chắc phải tìm cách khác chứ làm như thế này ko hay lắm

**Review by iTanChi — DISMISSED** [review]

**Review by iTanChi — APPROVED** [review]
